plugins/bundle/plugin.go

```diff
@@ -213,7 +216,11 @@ func (p *Plugin) Reconfigure(ctx context.Context, config interface{}) {
 
 	readyNow := p.ready
 
-	for name, source := range p.config.Bundles {
+	p.cfgMtx.RLock()
+	bundles := p.config.Bundles
+	p.cfgMtx.RUnlock()
+
```

Comment:
Currently the below operation is performed atomically. Do we know the consequences of changing that behavior?

Reply:
This needs to be changed (be atomic for the iteration). The config is read without the lock in the for statement.